script.js

```diff
@@ -1,12 +1,271 @@
+const drdEpisodes = document.getElementById("drdEpisodes");
+
+// ---  EVENT LISTENERS ----
+// Add event listeners for search box
+const inputBox = document.querySelector("#search_keyword");
+inputBox.addEventListener("input", (event) => {
+  console.log("search keyword : " + inputBox.value);
+  let keyword = inputBox.value.trim();
+  drdEpisodes.getElementsByTagName("option")[0].selected = "selected";
+  //inputBox.value.trim() == "" ? makePageforEpisodes() :
+  searchData(keyword);
+});
+
+// Add event listeners for Shows Select Box
+drdShows.addEventListener("change", (event) => {
+  currentShowId = drdShows.value;
+  findEpisodeByShowId(drdShows.value);
+});
+
+// Add event listeners for Episodes Select Box
+drdEpisodes.addEventListener("change", (event) => {
+  findEpisodeById(drdEpisodes.value);
+});
+//-------------------------------
+
+//-------- HEADER MAKERS --------------
+
+//--------------------------------------
+
+// Add to DropDown Lists ---------------
+
+function fill_ddList_Shows(showsdata) {
+  Object.entries(showsdata).forEach(function ([index, shows]) {
+    var opt = document.createElement("option");
+    opt.value = shows.id;
+    opt.text = shows.name;
+    drdShows.appendChild(opt);
+  });
+  drdShows.getElementsByTagName("option")[0].selected = "selected";
+}
+
+function fill_ddList_Episodes(showId) {
+  reset_drdEpisodes();
+  getEpisodesAsync(showId).then((data) =>
+    Object.entries(data).forEach(function ([index, e]) {
+      var selectedValue = episodeFormatter(e.season, e.number) + " - " + e.name;
+      var opt = document.createElement("option");
+      opt.value = e.id;
+      opt.text = selectedValue;
+      drdEpisodes.appendChild(opt);
+    })
+  );
+
+  //- makeEpisodesSelect(data)
+}
+
+function reset_drdEpisodes() {
+  drdEpisodes.innerText = "";
+  var opt = document.createElement("option");
+  opt.value = 0;
+  opt.text = "-- ALL EPISODES --";
+  drdEpisodes.appendChild(opt);
+  drdEpisodes.getElementsByTagName("option")[0].selected = "selected";
+}
+
+function makeEpisodesSelect(episode) {
+  var selectvalue =
+    episodeFormatter(episode.season, episode.number) + " - " + episode.name;
+  var opt = document.createElement("option");
+  opt.value = episode.id;
+  opt.text = selectvalue;
+  drdEpisodes.appendChild(opt);
+}
+
+//------------------------------
+
+//---- PAGE Makers ---------------
+function make_ShowsGrid(showsData) {
+  Object.entries(allShows).forEach(function ([index, e]) {
+    makeItem_Shows(e);
+  });
+}
+
+//---------------------------------
+function searchData(keyword) {
+  if (currentShowId != 0) {
+    getEpisodesAsync(currentShowId).then((e) => {
+      let searchResult = e.filter((episode) => {
+        return (
+          episode.name.toLowerCase().includes(keyword.toLowerCase()) ||
+          episode.summary.toLowerCase().includes(keyword.toLowerCase())
+        );
+      });
+      console.log("104 : " + keyword);
+      makePageforSearchedepisodes(searchResult);
```

Comment:
after you finished your code it is better to delete all console.log , arrange your codes